### PR TITLE
feat(aio): search result styling, footer and refactor

### PR DESCRIPTION
**Pull request by sjtrimble**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: aio/src/app/app.component.html [hunk 1]
```diff
@@ -8,13 +8,21 @@
...
   <section class="sidenav-content">
     <aio-search-results #searchResults></aio-search-results>
     <aio-doc-viewer [doc]="currentDocument | async" (docRendered)="onDocRendered($event)"></aio-doc-viewer>
+
+    <footer>
+      <div class="footer">
+        <p>Powered by Google ©2010-2017. Code licensed under an <a href="/license">MIT-style License</a>.</p>
+        <p>Documentation licensed under <a href="http://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>.
+        </p>
+        <p class="version-info">Version Info | {{ (versionInfo | async)?.full }}</p>
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: @sjtrimble did you agree this with @IgorMinar - Last I heard he was quite happy to have the version front and centre until we are closer to production.

File: aio/src/app/app.component.html [hunk 2]
```diff
@@ -8,7 +8,7 @@
 <md-sidenav-container class="sidenav-container">
 
   <md-sidenav #sidenav class="sidenav" [opened]="isSideBySide" [mode] = "isSideBySide ? 'side' : 'over'">
-    <aio-top-menu *ngIf="!isSideBySide" class="small" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu>
+    <aio-top-menu *ngIf="!isSideBySide" class="small" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu>
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: trailing whitespace 🙀

File: aio/src/styles/2-modules/_code.scss [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+    @include codeblock($backgroundgray);
     color: $darkgray;
-    padding: 20px;
-    margin: 24px auto;
-    border: 0.5px solid $lightgray;
 }
 
-.code-example pre {
+.code-shell code {
+    @include codeblock($darkgray);
```
Comment (petebacondarwin) [inline aio/src/styles/2-modules/_code.scss]: I think the padding and margins are a little off for this:

screen shot 2017-03-23 at 10 48 29

File: aio/src/styles/1-layouts/_search-results.scss [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+        font-size: 14px;
+        color: $lightgray;
         text-decoration: none;
+        &:hover {
+            color: $blue;
+            font-weight: 500;
+        }
+        &:visited {
+        text-decoration: none;
+        }
```
Comment (petebacondarwin) [inline aio/src/styles/1-layouts/_search-results.scss]: The search looks a lot nicer. We still need to do a bit of work with the padding maybe?

screen shot 2017-03-23 at 10 51 30

Also if the search is longer than the page then it is not possible to scroll down the results.

File: aio/src/styles/1-layouts/_footer.scss [hunk 5]
```diff
@@ -1,8 +1,8 @@
 footer .footer {
-    width: 100%;
     padding: 20px;
     z-index: 10;
-    background-color: $blue;
+    background-image: url('../src//assets/images/backgrounds/ng-pattern-sm.png');
+    background-repeat: repeat;
```
Comment (petebacondarwin) [inline aio/src/styles/1-layouts/_footer.scss]: Loving the new footer :-)

File: aio/src/app/app.component.html [hunk 6]
```diff
@@ -8,13 +8,21 @@
 <md-sidenav-container class="sidenav-container">
 
   <md-sidenav #sidenav class="sidenav" [opened]="isSideBySide" [mode] = "isSideBySide ? 'side' : 'over'">
-    <div class="version-info vertical-menu-item">{{ (versionInfo | async)?.full }}</div>
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: perhaps we should leave this in place until we get close to production?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
This looks very nice!
I made some minor comments that can be improved in later PRs.
I think we need to check about the version info thing with @IgorMinar before we change it.

File: aio/src/app/app.component.html (hunk 6 above)
Comment (wardbell) [inline aio/src/app/app.component.html]: @petebacondarwin Stef moved it to the footer (ln 25) and Igor is cool with that.

Comment (wardbell) [inline aio/src/app/app.component.html]: I'm seeing a blue artifact like this on the mini toolbar.

Also, IMO, we should have a tinier logo ... or, better, none at all here ... and bigger font for the text.
mini-nav-toolbar

File: aio/src/styles/2-modules/_code.scss (hunk 3 above)
Comment (wardbell) [inline aio/src/styles/2-modules/_code.scss]: Why are we showing ANY headers in ALL CAPS.  Have your suggestions on h1-h5 not landed yet?

File: aio/src/styles/mixins.scss [hunk 7]
```diff
@@ -6,4 +6,12 @@
   transform: rotate($degrees);
   transition: transform 150ms;
   transition-timing-function: ease-in-out;
+}
+
+@mixin codeblock($bgcolor) {
```
Comment (wardbell) [inline aio/src/styles/mixins.scss]: Remind me to factor this into our revised syntax highlighting PR

**Review by wardbell — COMMENTED** [review]
My feedback. Not withholding approval because my suggestions could come in a follow up PR. Want to keep moving forward !

File: aio/src/styles/1-layouts/_footer.scss (hunk 5 above)
Comment (wardbell) [inline aio/src/styles/1-layouts/_footer.scss]: Me too. I think it needs some whitespace above it
image

**Comment by sjtrimble** [comment]
1. I just fixed the last commit message to follow the guidelines.

2. You don't want headers in caps? That was part of my style summary we all reviewed a while back. I can change it if you think it's not appropriate. I have noticed that often h3s are used when h2s could be used, which does change the look and feel of things.

3. @wardbell I don't see the blue outline on the mini toolbar, but I do see something similar. I can look into it some more (I was going to ask you if you knew what that was).

4. I will push in another commit shortly for some of the other smaller items you mentioned.

5. @petebacondarwin I did check about the version number with Igor, which is why I moved it.

@wardbell This is the goal for the side nav menu when collapsed. @petebacondarwin said he would help with the logic on this one. Only change would be the the logo should be just the badge icon, not the full logo. This would address one of your comments.

screen shot 2017-03-22 at 11 50 02 pm

**Review by petebacondarwin — APPROVED** [review]

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**